sql/parser/normalize.go

```diff
@@ -353,10 +353,17 @@ var _ Visitor = &isConstVisitor{}
 
 func (v *isConstVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
 	if pre && v.isConst {
-		switch expr.(type) {
+		switch t := expr.(type) {
 		case DReference, *ExistsExpr, *QualifiedName, *Subquery, ValArg:
 			v.isConst = false
 			return nil, expr
+		case *FuncExpr:
+			// We need to type check this guy to make sure `fn` is populated.
+			_, _ = typeCheckFuncExpr(t)
```

Comment:
Should you be ignoring the error here? Seems like `t.fn` won't be filled in correctly in which case I think you should set `v.isConst = false`.